src/main/resources/db_scripts/migrations/2025-10-questions-attempts-add-mark-field.sql

```diff
@@ -0,0 +1,99 @@
+ALTER TABLE question_attempts
+ADD marks integer;
+
+ALTER TABLE quiz_question_attempts
+ADD marks integer;
+
+CREATE OR REPLACE FUNCTION update_marks_for_period(qa question_attempts)
+    RETURNS int AS $$
+BEGIN
+    RETURN CASE
+            WHEN (qa.question_attempt -> 'answer' ->> 'type') = 'llmFreeTextChoice'
+                THEN (qa.question_attempt ->> 'marksAwarded')::int
+                ELSE qa.correct::int
+        END;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION quiz_update_marks_for_period(qa quiz_question_attempts)
+    RETURNS int AS $$
+BEGIN
+    RETURN CASE
+               WHEN (qa.question_attempt -> 'answer' ->> 'type') = 'llmFreeTextChoice'
+                   THEN (qa.question_attempt ->> 'marksAwarded')::int
+               ELSE qa.correct::int
+        END;
+END;
+$$ LANGUAGE plpgsql;
+
+UPDATE question_attempts SET marks = update_marks_for_period(question_attempts) WHERE timestamp > '2014-01-01' AND timestamp <= '2015-01-02';
```

Comment:
Are these groupings based on the ones I did the last time we re-wrote `question_attempts`? I was wondering where they came from 😅